# HG changeset patch
# User Tatsuro Matsuoka
# Date 1207275125 14400
#      Thu Apr 03 22:12:05 2008 -0400
# Branch release-3-0-x
# Node ID b5bbcaa70105c9de3afac43fa90a044914bd4467
# Parent  5892d5533b2ba24fd8796226797cf70eb3c050cb
update README.Cygwin

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,12 @@
+2008-04-03  Tatsuro MATSUOKA  <tmacchant@yahoo.co.jp>
+
+	* README.Cygwin: Update.
+
 2008-03-11  David Bateman  <dbateman@free.fr>
 
 	* run-octave.in: Fix typo.
 
 2008-02-21  John W. Eaton  <jwe@octave.org>
 
 	* examples/Makefile.in (install install-strip): Ignore errors
 	installing octave.desktop file.
diff --git a/README.Cygwin b/README.Cygwin
--- a/README.Cygwin
+++ b/README.Cygwin
@@ -7,15 +7,52 @@ recommended that you use a more recent v
 It should be possible to build Octave on Windows systems with Cygwin,
 but at the time of this writing, there are some performance problems
 related to the way C++ exception handling is implemented with the
 default Cygwin compiler.  This is a known problem with a long history.
 If you would like to see this problem corrected, please search the
 Cygwin mailing lists for threads related to "sjlj exception handling"
 (or similar).
 
+There are also two "unofficial" Octave distributions for Cygwin:
+
+ 1. http://www.geocities.jp/tmacchant
+
+    The binaries here are built using gcc-3.4.4-3 configured with
+    --disable-sjlj-exceptions.  Performance is improved by using DWARF
+    exception handling instead of setjump/longjump exception
+    handling.  However, to build dynamically loaded .oct files that
+    will work with this version of Octave, you must use the same
+    specially configured version of GCC that was used to build Octave
+    itself and not the version of GCC that is distributed with
+    Cygwin.
+
+    This binary is maintained by Tatsuro Matsuroka.
+
+
+ 2. http://matzeri.altervista.org
+
+    The binaries here aim to be an officail cygwin distribution of
+    Octave-3.0.x and are built using the version of GCC distributed
+    with Cygwin.  Performance of linear algebra functions is fine, but
+    the performance of the scripting language interpreter suffers
+    because of the setjump/longjump exception handling model used in
+    the version of GCC distributed with Cygwin.  The advantage is that
+    you don't need a special version of GCC.
+
+    This binary is maintained by Marco Atzeri
+
+We hope that Cygwin will eventually have a version of GCC that does
+not suffer from the performance problem related to setjump/longjump
+exception handling.
+
 
 John W. Eaton
 jwe@bevo.che.wisc.edu
 University of Wisconsin-Madison
 Department of Chemical & Biological Engineering
 
-Last updated: Wed, 31 Oct 2007 16:21:43 EDT
+Tatsuro MATSUOKA 
+tmacchant@yahoo.co.jp
+Department of Molecular Design and Engineering, 
+Gradudate School of Engineering, Nagoya University.
+
+Last updated: Thu, 03 Apr 2008 22:01:48 EDT
